packages/types/src/transaction/perspective.ts

```diff
@@ -43,6 +47,20 @@ export const viewActionFromEmptyPerspective = (action: Action): ActionView | und
           }),
         },
       });
+    case 'swap':
+      return new ActionView({
+        actionView: {
+          case: 'swap',
+          value: new SwapView({
+            swapView: {
+              case: 'opaque',
+              value: new SwapView_Opaque({
+                swap: action.action.value,
+              }),
+            },
+          }),
+        },
+      });
```

Comment:
i guess this matches other sections of this file, but i'd like to point out you don't have to nest constructors like this - it's fine to just use the top-level type

```suggestion
    case 'swap':
      return new ActionView({
        actionView: {
          case: 'swap',
          value: {
            swapView: {
              case: 'opaque',
              value: {
                swap: action.action.value,
              },
            },
          },
        },
      });
```